tests/unit/test_charm.py

```diff
@@ -25,9 +25,20 @@ def setup_postgres_relation(harness):
     )
 
 
-def test_update_container_correct_config(
-    harness, mocked_kubernetes_service_patcher, mocked_sql_migration
-) -> None:
+def trigger_database_changed(harness) -> None:
+    db_relation_id = harness.add_relation("pg-database", "postgresql-k8s")
+    harness.add_relation_unit(db_relation_id, "postgresql-k8s/0")
+    harness.update_relation_data(
+        db_relation_id,
+        "postgresql-k8s",
+        {
+            "data": '{"database": "database", "extra-user-roles": "SUPERUSER"}',
+            "endpoints": DB_ENDPOINTS,
+        },
+    )
+
+
+def test_correct_config(harness, mocked_kubernetes_service_patcher, mocked_sql_migration) -> None:
     harness.begin()
```

Comment:
You can place the `begin` in a setUp it's probably used in each test

Reply:
The next test asserts the empty initial pebble plan before beginning